### DISCUSSION TITLE
wazuh-agent and wash-monitoring ilm / index lifecycle management

### DISCUSSION THREAD

**Discussion by pgaufillet** [wazuh/wazuh#14364] [Q&A] [answered]
Hi,

while it is possible to set up index lifecycle management for wazuh-alerts/wazuh-archives, I don't find any information about setting it up for wazuh-monitoring/wazuh-statistics indexes. Has anyone a clue about how to proceed?

**Comment by mdiego92** [comment, marked as answer]
Hello pgaufillet,

We have a complete article in our blog with all the details to create and set up index lifecycle policies:
https://wazuh.com/blog/wazuh-index-management/

Please let us know if you have any questions or issues regarding this guide and we will answer them back!

Thank you!

**Reply by pgaufillet** [reply]
Thank you. Indeed this is the article that I have used for setting it up for alerts and archives. But I don't know where the index templates for monitoring and statistics are, and associating manually them in Elasticsearch only lasts for a while: It looks the templates are sometimes automatically renewed.

**Reply by mdiego92** [reply]
Hello pgaufillet,

To enable, disable or modify the statistics and monitoring indices, you could go to **Wazuh** > **Settings** > **Configuration**:

![image](https://user-images.githubusercontent.com/62477966/180301857-9dd8fd8f-3833-4140-94ed-120b46369795.png)

To configure the indices you will need to set up a new policy for statistics and another one for monitoring in **OpenSearch Plugins** > **Index policies**. When you create or edit, you're able to set up the index pattern to apply the policy:

![image](https://user-images.githubusercontent.com/62477966/180302375-6010bbd4-bea5-4412-8538-367e8abf7990.png)

Please let us know if you have any remaining questions!

Thank you very much.

**Reply by pgaufillet** [reply]
Thank you for the explanation @mdiego92. 

Nevertheless, it doesn't offer any ILM option or solution to my issue (even if useful and already used as for the index creation parameters). Coming back to the initial report, it looks the wazuh-monitoring and wazuh-statistics index templates are periodically rewritten. When it happens, the associated ILM policy is lost. The only workaround found up to now is to associate manually each month the new indexes with the appropriate ILM policy, which is obviously neither user friendly nor efficient. 

Other suggestions welcome   :-)

**Reply by mdiego92** [reply]
Hello pgaufillet,

Could you please share with us your ILM policy template to further analyze your request?

Thank you very much and sorry for the late response!

**Reply by pgaufillet** [reply]
Hello @mdiego92,

no problem: I greatly appreciate your help :-)

Here is the ILM policy we are talking about (extracted with `GET _ilm/policy/wazuh-retention-policy`):
 
```
"wazuh-retention-policy" : {
    "version" : 5,
    "modified_date" : "2022-08-06T08:54:58.888Z",
    "policy" : {
      "phases" : {
        "hot" : {
          "min_age" : "0ms",
          "actions" : { }
        },
        "delete" : {
          "min_age" : "365d",
          "actions" : {
            "delete" : {
              "delete_searchable_snapshot" : true
            }
          }
        }
      }
    },
    "in_use_by" : {
      "indices" : [
        "wazuh-alerts-4.x-2022.08",
        "wazuh-statistics-2021.11",
        "wazuh-statistics-2021.12",
        "wazuh-alerts-4.x-2022.04",
        "wazuh-monitoring-2022.01",
        "wazuh-alerts-4.x-2022.03",
        "wazuh-monitoring-2022.02",
        "wazuh-alerts-4.x-2022.06",
        "wazuh-alerts-4.x-2022.05",
        "wazuh-alerts-4.x-2022.08",
        "wazuh-alerts-4.x-2022.07",
        "wazuh-monitoring-2022.07",
        "filebeat-7.12.1-2021.12",
        "filebeat-7.12.1-2021.11",
        "wazuh-monitoring-2022.05",
        "wazuh-monitoring-2022.06",
        "wazuh-monitoring-2022.03",
        "wazuh-alerts-4.x-2022.02",
        "wazuh-alerts-4.x-2022.01",
        "wazuh-monitoring-2022.04",
        "wazuh-statistics-2022.05",
        "wazuh-statistics-2022.04",
        "wazuh-statistics-2022.07",
        "wazuh-statistics-2022.06",
        "wazuh-statistics-2022.01",
        "wazuh-statistics-2022.03",
        "wazuh-statistics-2022.02",
        "wazuh-monitoring-2021.11",
        "wazuh-monitoring-2021.12",
        "filebeat-7.12.1-2022.01",
        "filebeat-7.12.1-2022.02",
        "filebeat-7.12.1-2022.05",
        "wazuh-alerts-4.x-2021.11",
        "filebeat-7.12.1-2022.03",
        "wazuh-alerts-4.x-2021.12",
        "filebeat-7.12.1-2022.04",
        "filebeat-7.17.4-2022.07",
        "filebeat-7.14.2-2022.05",
        "filebeat-7.14.2-2022.07"
      ],
      "data_streams" : [ ],
      "composable_templates" : [ ]
    }
  }
```

But honestly I don't think the problem is coming from it. It is very simple and works perfectly for wash-alerts and filebeat indexes.

**Reply by mdiego92** [reply]
Hello pgaufillet,

In this case, I would need you to send me the JSON of your "wazuh-retention-policy" policy. You will need to select the policy and copy the content of the "View JSON" section:

![image](https://user-images.githubusercontent.com/62477966/184441396-b263cb4e-436f-4d2a-a861-9f2be3e8c234.png)
 
Thank you!

**Reply by pgaufillet** [reply]
Hello @mdiego92,

As our exchange has not concluded, because this issue is still present, and because I am currently improving my deployment of Wazuh ;-), I would like to reactivate this conversation. 

Is the JSON file I have sent what you were expecting? Do you need something else? Any piece of news since last Summer?

**Reply by mdiego92** [reply]
Hello @pgaufillet,

Last year I requested to you the configuration of your policy in JSON format. You can check it from the UI (Index Management section) without having to execute the API.

Please let me know if you're still having issues. Happy to help you with this!
